core/sql/optimizer/ScanOptimizer.cpp

```diff
@@ -8590,6 +8595,7 @@ void MDAMCostWA::compute()
 // 	  // invalidate the cache
 // 	  disjunctsFR_.reset();
 // 	  disjunctsLR_.reset();
+          MDAM_DEBUG0(MTL2, "Mdam scan lost due to exceeding cost bound");
```

Comment:
Yes, these debug messages can help with narrow down the problem quickly.

On Fri, Oct 23, 2015 at 12:48 PM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/optimizer/ScanOptimizer.cpp
> <https://github.com/apache/incubator-trafodion/pull/144#discussion_r42893715>
> :
>
> > @@ -8590,6 +8595,7 @@ void MDAMCostWA::compute()
> >  // 	  // invalidate the cache
> >  // 	  disjunctsFR_.reset();
> >  // 	  disjunctsLR_.reset();
> > +          MDAM_DEBUG0(MTL2, "Mdam scan lost due to exceeding cost bound");
>
> I LOVE this kind of debugging aid.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/144/files#r42893715>.
>



-- 
Regards, --Qifan